src/handlers/generate-erc20-permit.ts

```diff
@@ -1,41 +1,77 @@
 import { PERMIT2_ADDRESS, PermitTransferFrom, SignatureTransfer } from "@uniswap/permit2-sdk";
 import { ethers, keccak256, MaxInt256, parseUnits, toUtf8Bytes } from "ethers";
-import { Context } from "../types/context";
+import { Context, Logger } from "../types/context";
 import { Permit } from "../types/permits";
 import { decryptKeys } from "../utils/keys";
 import { getPayoutConfigByNetworkId } from "../utils/payoutConfigByNetworkId";
 
-export async function generateErc20PermitSignature(context: Context, username: string, amount: number): Promise<Permit> {
-  const config = context.config;
-  const logger = context.logger;
-  const { evmNetworkId, evmPrivateEncrypted } = config;
-  const { user, wallet } = context.adapters.supabase;
+export async function generateErc20PermitSignature(
+  username: string,
+  amount: number,
+  evmNetworkId: number,
+  evmPrivateEncrypted: string,
+  userId: number,
+  walletAddress: string,
+  issueId: number,
+  logger: Logger
+): Promise<Permit>;
+export async function generateErc20PermitSignature(username: string, amount: number, context: Context): Promise<Permit>;
+export async function generateErc20PermitSignature(
+  username: string,
+  amount: number,
+  contextOrNetworkId: Context | number,
```

Comment:
Wouldn't it be better to use union type here?
Doing it like this reduces type safety because you can still put a number as `contextOrNetworkId` but specify `undefined` as `userId`.

Reply:
@whilefoo Yes I intend to have another type to put something like `Context | Payload` which would solve a previous comment by @gitcoindev on the order of the arguments, and also solve type safety issues. I am doing that in another PR [here](https://github.com/ubiquibot/permit-generation/pull/10)if that's ok, because I had to change lots of other types as well.